backend/src/db-utils.js

```diff
@@ -120,6 +84,108 @@ const queryOnlineNodes = async () => {
+    } else {
+      query = `SELECT gas_price FROM blocks ORDER BY height DESC LIMIT 1`;
+    }
+    return await querySingleRow([query], { dataSource });
+  }
+  async function queryLastMinuteBlocks({ dataSource }) {
+    let query;
+    if (dataSource === DS_INDEXER_BACKEND) {
+      query = `SELECT block_timestamp AS latest_block_timestamp FROM blocks ORDER BY block_timestamp DESC LIMIT 1`;
+    } else {
+      query = `SELECT timestamp AS latest_block_timestamp FROM blocks ORDER BY timestamp DESC LIMIT 1`;
+    }
+    const latestBlockTimestampOrNone = await querySingleRow([query], {
+      dataSource,
+    });
+    if (!latestBlockTimestampOrNone) {
+      return { total: 0 };
+    }
+    const {
+      latest_block_timestamp: latestBlockTimestamp,
+    } = latestBlockTimestampOrNone;
+    const latestBlockTimestampBN = new BN(latestBlockTimestamp);
+    const nowEpochTimeBN = new BN(Math.floor(new Date().getTime() / 1000));
+    let latestBlockEpochTimeBN;
+    if (dataSource == DS_INDEXER_BACKEND) {
+      latestBlockEpochTimeBN = latestBlockTimestampBN.div(new BN("1000000000"));
+    } else {
+      latestBlockEpochTimeBN = latestBlockTimestampBN.divn(1000);
+    }
+    // If the latest block is older than 1 minute from now, we report 0
+    if (nowEpochTimeBN.sub(latestBlockEpochTimeBN).gtn(60)) {
+      return { total: 0 };
+    }
+
+    if (dataSource === DS_INDEXER_BACKEND) {
+      query = `SELECT COUNT(*) AS total FROM blocks
+        WHERE block_timestamp > (:latestBlockTimestamp - 60 * 1000000000)`;
+    } else {
+      query = `SELECT COUNT(*) AS total FROM blocks
+        WHERE timestamp > (:latestBlockTimestamp - 60 * 1000)`;
+    }
+    return await querySingleRow([query, { latestBlockTimestamp }], {
+      dataSource,
+    });
+  }
+  const [
+    lastBlockHeight,
+    latestGasPrice,
+    lastMinuteBlocks,
+  ] = await Promise.all([
+    queryLatestBlockHeight(options),
+    queryLatestGasPrice(options),
+    queryLastMinuteBlocks(options),
+  ]);
+  return {
+    latestBlockHeight: lastBlockHeight ? lastBlockHeight.block_height : 0,
+    latestGasPrice: latestGasPrice.gas_price,
+    numberOfLastMinuteBlocks: lastMinuteBlocks.total,
+  };
+};
+
+const queryDashboardTxInfo = async (options) => {
+  async function queryLastDayTxCount({ dataSource }, day = 1) {
+    let query;
+    if (dataSource === DS_INDEXER_BACKEND) {
+      query = `SELECT COUNT(*) AS total FROM transactions
+        WHERE block_timestamp > (EXTRACT(EPOCH FROM NOW()) - 60 * 60 * 24 * ${day}) * 1000000000`;
+    } else {
+      query = `SELECT COUNT(*) AS total FROM transactions
+        WHERE block_timestamp > (strftime('%s','now') - 60 * 60 * 24 * ${day}) * 1000`;
+    }
+    return await querySingleRow([query], { dataSource });
+  }
+
+  let transactionCountArray = new Array(14);
+  for (let i = 0; i < transactionCountArray.length; i++) {
+    let result = await queryLastDayTxCount(options, i + 1);
```

Comment:
This is super inefficient to do queries in a loop. Try it out with the Indexer database (it might be fine for you with the old backend since you don't have the full database, but it is as bad as on the indexer Postgres database).

Select the transactions which are less than 15 days old, then group by the day (timestamp divided by the amount of milliseconds/nanoseconds in 24 hours) - all that in a single query